### PR TITLE
CASMINST-4161 adjust interface mac bond check test for optional can0

### PR DESCRIPTION
**Pull request by jacobsalmela**
Signed-off-by: Jacob Salmela <jacob.salmela@hpe.com>

## Summary and Scope

- Adjusts `check_interface_mac_matches_bond.sh` to accept multiple arguments to facilitate checking them all in the same script and removes logging in favor of the bullet below
- Adjusts `goss-interfaces-have-same-mac.yaml` to use `log_run.sh`
- Adjusts `goss-interfaces-have-same-mac.yaml` to utilize a small loop while retaining functionality of the `.Env.GOSS_BASE` variable since that variable cannot be evaluated in the current loop of `network_interfaces`.
- Fixes a mismatched quote in `ncn_healthcheck` (opening `"` with a closing `'`) introduced with #220 

## Issues and Related PRs

- CASMINST-4161
- Relates to CASMINST-4183

## Testing

Tested manually

### Tested on:

  * `surtur`
  * `wasp`

### Test description:

on wasp **without** `can0`

```
wasp-ncn-m001-pit:~ # goss --vars /opt/cray/tests/install/ncn/vars/variables-ncn.yaml validate
.

Total Duration: 0.043s
Count: 1, Failed: 0, Skipped: 0
wasp-ncn-m001-pit:~ # cat /opt/cray/tests/install/logs/check_mac/20220308_202347_297618660_22208.log
log_run argument(s): -l check_mac
Executable: '/opt/cray/tests/install/ncn//scripts/check_interface_mac_matches_bond.sh'
4 argument(s) to executable: 'bond0.nmn0' 'bond0.hmn0' 'bond0.can0' 'bond0.cmn0'

## 20220308_202347_306114781 ##                      executable starting ##
###########################################################################
Checking that MAC address of interface 'bond0.nmn0' matches MAC address of bond0...

b8:59:9f:f9:1b:fe (bond0)
b8:59:9f:f9:1b:fe (bond0.nmn0)
Checking that MAC address of interface 'bond0.hmn0' matches MAC address of bond0...

b8:59:9f:f9:1b:fe (bond0)
b8:59:9f:f9:1b:fe (bond0.hmn0)
Checking that MAC address of interface 'bond0.can0' matches MAC address of bond0...

Device "bond0.can0" does not exist.
bond0.can0 is optional...skipping
###########################################################################
## 20220308_202347_331642266 ##       executable completed (exit code=0) ##
```

on surtur **with** `can0`

```
surtur-ncn-m001-pit:~ # export GOSS_BASE=/opt/cray/tests/install/ncn/
surtur-ncn-m001-pit:~ # export GOSS_FILE=/opt/cray/tests/install/ncn/tests/goss-interfaces-have-same-mac.yaml
surtur-ncn-m001-pit:~ # goss --vars /opt/cray/tests/install/ncn/vars/variables-ncn.yaml validate
.

Total Duration: 0.048s
Count: 1, Failed: 0, Skipped: 0
surtur-ncn-m001-pit:~ # cat /opt/cray/tests/install/logs/check_mac/20220308_204245_858191616_19789.log
log_run argument(s): -l check_mac
Executable: '/opt/cray/tests/install/ncn//scripts/check_interface_mac_matches_bond.sh'
4 argument(s) to executable: 'bond0.nmn0' 'bond0.hmn0' 'bond0.can0' 'bond0.cmn0'

## 20220308_204245_867189864 ##                      executable starting ##
###########################################################################
Checking that MAC address of interface 'bond0.nmn0' matches MAC address of bond0...

14:02:ec:da:b9:98 (bond0)
14:02:ec:da:b9:98 (bond0.nmn0)
Checking that MAC address of interface 'bond0.hmn0' matches MAC address of bond0...

14:02:ec:da:b9:98 (bond0)
14:02:ec:da:b9:98 (bond0.hmn0)
Checking that MAC address of interface 'bond0.can0' matches MAC address of bond0...

14:02:ec:da:b9:98 (bond0)
14:02:ec:da:b9:98 (bond0.can0)
Checking that MAC address of interface 'bond0.cmn0' matches MAC address of bond0...

14:02:ec:da:b9:98 (bond0)
14:02:ec:da:b9:98 (bond0.cmn0)
###########################################################################
## 20220308_204245_897458768 ##       executable completed (exit code=0) ##
```

## Risks and Mitigations

Medium.  This changes the script slightly to accept any arguments and then loops through them, using the existing logic.   It also removes the logging component in favor of the `log_run.sh` script in #220 

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [ ] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: goss-testing/scripts/check_interface_mac_matches_bond.sh [hunk 1]
```diff
@@ -61,67 +51,49 @@ Exits 0 if this is the case. Exits non-0 if not, or if there are any errors.
...
-[[ -n $INT_MAC ]] || error 70 "No MAC address found for $INTERFACE"
-[[ -n $BND_MAC ]] || error 75 "No MAC address found for bond0"
-
-# Finally, validate that they match
-[[ $INT_MAC == $BND_MAC ]] || error 90 "MAC address of $INTERFACE does not match MAC address of bond0"
+# if the interface does not show, it might be the option CAN
+if ! eval ip addr show dev "$INTERFACE" 1>/dev/null; then
+  if [[ "$INTERFACE" == "bond0.can0" ]]; then
+    echo "$INTERFACE is optional...skipping"
+    # so we can skip the rest of the loop here
```
Comment (heemstra) [inline goss-testing/scripts/check_interface_mac_matches_bond.sh:76]: What I'm wondering here is whether the test should know whether it's supposed to be present or not... vs. simply not caring if it's not.

**Review by mharding-hpe — APPROVED** [review]

File: goss-testing/scripts/check_interface_mac_matches_bond.sh (hunk 1 above)
Comment (jacobsalmela) [reply goss-testing/scripts/check_interface_mac_matches_bond.sh:76]: @trad511 any input?

Comment (mharding-hpe) [reply goss-testing/scripts/check_interface_mac_matches_bond.sh:76]: I agree that would be the ideal solution, but this PR is a good short term solution, IMO.

**Comment by jacobsalmela** [comment]
tagged and rebased

File: goss-testing/scripts/check_interface_mac_matches_bond.sh [hunk 2]
```diff
@@ -61,67 +51,49 @@ Exits 0 if this is the case. Exits non-0 if not, or if there are any errors.
...
 echo
 
-# Validate that we actually got a MAC address for each
-[[ -n $INT_MAC ]] || error 70 "No MAC address found for $INTERFACE"
-[[ -n $BND_MAC ]] || error 75 "No MAC address found for bond0"
-
-# Finally, validate that they match
-[[ $INT_MAC == $BND_MAC ]] || error 90 "MAC address of $INTERFACE does not match MAC address of bond0"
+# if the interface does not show, it might be the option CAN
+if ! eval ip addr show dev "$INTERFACE" 1>/dev/null; then
```
Comment (mharding-hpe) [inline goss-testing/scripts/check_interface_mac_matches_bond.sh:73]: I don't think you need the "eval" here, but equally I don't think it hurts anything.

**Review by mharding-hpe — APPROVED** [review]

**Comment by mharding-hpe** [comment]
@jacobsalmela I cherry-picked your mismatched quote fix into a separate PR, since this one does not appear to be merging, but we need to get that fix merged. That PR is:
https://github.com/Cray-HPE/csm-testing/pull/230

**Review by johren-hpe — APPROVED** [review]

**Pull request merged**